freqtrade/persistence/trade_model.py

```diff
@@ -1296,6 +1327,90 @@ class Trade(ModelBase, LocalTrade):
     funding_fees: Mapped[Optional[float]] = mapped_column(
         Float(), nullable=True, default=None)  # type: ignore
 
+    @hybrid_property
+    def open_orders(self):
+        return [order for order in self.orders if order.ft_is_open]
+
+    @open_orders.expression
+    def open_orders(cls):
+        return (
+            select(Order).where(Order.ft_is_open is True)
+            .where(
+                Order.order_id.in_(
+                    select(Order.order_id)
+                    .where(Order.ft_trade_id == cls.id)
+                )
+            )
+        )
+
+    @hybrid_property
+    def has_open_orders(self) -> int:
+        open_orders_wo_sl = []
+        for oo in self.open_orders:
+            if (oo.ft_order_side not in ['stoploss']):
+                open_orders_wo_sl.append(oo)
+
+        return (len(open_orders_wo_sl) > 0)
+
+    @has_open_orders.expression
+    def has_open_orders(cls) -> int:
+        return (
+            select(func.exists())
+            .where(Order.ft_is_open.is_(True))
+            .where(Order.ft_order_side != "stoploss")
+            .where(Order.ft_trade_id == cls.id)
+            .as_scalar()
+        )
```

Comment:
I can't think of a case where it's faster to go through the database than to do a in-memory list comprehension to get the same result.

as mentioned before  - `.orders` is a list that's filled in all cases when querying the trade from the database.

Reply:
Fixing that today, I'm suspecting this causing some of the the 15 failing tests on test_freqtradebot